### PR TITLE
add feature flags on start

### LINKED ISSUE #294: [featureflag] UI fails to load on M1 Mac - (Postgrex.Error) ERROR 42P01 (undefined_table) relation "featureflags" does not exist
# Bug Report

## Which version of the demo you are using? 
main/latest 
commit 1a4698f958622a798122fda3ac532938c6a93128

## Environment:
Kernel Version 21.6.0
RELEASE_ARM64_T6000 arm64
Docker Desktop 4.8.1

## Symptom

Feature flag UI doesn't load: "Internal Server Error"
http://localhost:8081/

Each reload produces the following log:
```
10:13:08.686 request_id=FwqRUMKG7L0YbksAAARB [info] GET /
10:13:08.693 request_id=FwqRUMKG7L0YbksAAARB [info] Sent 500 in 7ms
10:13:08.693 [error] #PID<0.2009.0> running FeatureflagserviceWeb.Endpoint (connection #PID<0.2008.0>, stream id 1) terminated
Server: localhost:8081 (http)
Request: GET /
** (exit) an exception was raised:
    ** (Postgrex.Error) ERROR 42P01 (undefined_table) relation "featureflags" does not exist

    query: SELECT f0."id", f0."description", f0."enabled", f0."name", f0."inserted_at", f0."updated_at" FROM "featureflags" AS f0
        (ecto_sql 3.8.2) lib/ecto/adapters/sql.ex:932: Ecto.Adapters.SQL.raise_sql_call_error/1
        (ecto_sql 3.8.2) lib/ecto/adapters/sql.ex:847: Ecto.Adapters.SQL.execute/6
        (ecto 3.8.3) lib/ecto/repo/queryable.ex:221: Ecto.Repo.Queryable.execute/4
        (ecto 3.8.3) lib/ecto/repo/queryable.ex:19: Ecto.Repo.Queryable.all/3
        (featureflagservice 0.1.0) lib/featureflagservice_web/controllers/page_controller.ex:7: FeatureflagserviceWeb.PageController.index/2
        (featureflagservice 0.1.0) lib/featureflagservice_web/controllers/page_controller.ex:1: FeatureflagserviceWeb.PageController.action/2
        (featureflagservice 0.1.0) lib/featureflagservice_web/controllers/page_controller.ex:1: FeatureflagserviceWeb.PageController.phoenix_controller_pipeline/2
        (phoenix 1.6.9) lib/phoenix/router.ex:354: Phoenix.Router.__call__/2
10:13:08.711 request_id=FwqRUMQIP1HPNQQAAASB [info] GET /null
10:13:08.711 request_id=FwqRUMQIP1HPNQQAAASB [info] Sent 404 in 314µs
```

## Reproduce

$ docker compose up
http://localhost:8081/

## Additional Context

Fails only on M1 Mac. Seems to be working fine on Ubuntu 22.04 LTS, x86_64.

### LINKED ISSUE #301: Create migration for feature flags introduced by #245
#245 introduces a new feature flag and needs a migration in the FF service set up to create it on launch.

### PR DESCRIPTION
**Pull request by puckpuck**
Fixes #301 

## Changes

- Inserts Feature Flag records for a productcatalog and shipping service failure modes

Inserting the records into the database using Ecto migrations with a default of `false` for enabled. The user can enable the flag by going to the Feature Flag service web UI http://localhost:8081. As of the creation of this PR only the `productCatalogFailure` flag is leveraged in service code. Using migrations means feature flags state will be preserved if you stop/restart the service/demo. The migrations only reset if the `postgres_ffs` service container is recreated with docker compose (using `docker compose down` or the `--force-recreate` flag)

We should merge #315 before this

* [x] Appropriate `CHANGELOG.md` updated for non-trivial changes

### CODE REVIEW

**Comment by puckpuck** [comment]
@tsloughter Can you offer guidance on how to get Ecto Migrations to wait for a database connection before proceeding instead of doing the `Process.sleep(5000)`?

**Comment by tsloughter** [comment]
I can't but I can find someone who can.

Would be nicer if the docker `depends_on` just did what one would expect. Actually, yea, I bet just a different postgres image may fix the issue (or a custom one). I'll also look into that, a postgres container that will not be considered started until its accessible.

**Comment by tsloughter** [comment]
@puckpuck can you try out https://github.com/open-telemetry/opentelemetry-demo/pull/315

**Comment by puckpuck** [comment]
ooof... I got caught mid-docs restructuring on this one.  Sorry for the commits reverting back and forth there.

File: docs/feature_flags.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Feature Flags
+
+This demo comes with 2 feature flags which control failure conditions in the
+Product Catalog and Shipping services. By default the flags are disabled. Using
```
Comment (mic-max) [inline docs/feature_flags.md:4]: This first sentence would need to change with the addition of more feature flags so I think we can remove it

File: src/featureflagservice/priv/repo/migrations/20220524172636_create_featureflags.exs [hunk 2]
```diff
@@ -11,5 +11,24 @@ defmodule Featureflagservice.Repo.Migrations.CreateFeatureflags do
...
+      description: "Fail shipping service when shipping a product to a non-USA address",
+      enabled: false})
+  end
+
+  defp execute_down do
+    repo().delete(%Featureflagservice.FeatureFlags.FeatureFlag{name: "productCatalogFailure"})
+    repo().delete(%Featureflagservice.FeatureFlags.FeatureFlag{name: "shippingFailure"})
   end
-end
+end
```
Comment (mic-max) [inline src/featureflagservice/priv/repo/migrations/20220524172636_create_featureflags.exs:34]: nit: \n EOF

**Review by mic-max — APPROVED** [review]
LGTM

**Review by mviitane — APPROVED** [review]
LGTM

**Review by reyang — APPROVED** [review]

**Pull request merged**